irohad/main/impl/on_demand_ordering_init.hpp

```diff
@@ -75,6 +75,7 @@ namespace iroha {
           consensus::Round initial_round,
           std::function<std::chrono::milliseconds(
               const synchronizer::SynchronizationEvent &)> delay_func,
+          size_t max_number_of_transactions,
```

Comment:
Please make this variable name consistent with `max_size` in `initOrderingGate`, maybe by renaming `max_size` to `max_number_of_transactions`.